### PR TITLE
TextPredictorGroupSpec

### PR DESCRIPTION
**Pull request by sarakolding**
## Notes for reviewers
Attempted to make a group spec for the TextPredictorSpec, since our feature gen scripts are currently very long and messy 🙃

### CODE REVIEW

**Comment by sarakolding** [comment]
imports are failing all of a sudden, and i'm so confused?

File: src/timeseriesflattener/feature_spec_objects.py [hunk 1]
```diff
@@ -998,3 +998,70 @@ def create_combinations(self) -> List[OutcomeSpec]:
...
+        allowed_nan_value_prop (List[float]):
+            If NaN is higher than this in the input dataframe during
+            resolution, raise an error. Defaults to: [0.0].
+        prefix (str):
+            Prefix for column name, e,g, <prefix>_<feature_name>. Defaults to: pred.
+        loader_kwargs (Optional[List[Dict[str, Any]]]):
+            Optional kwargs for the values_loader.
+        lookbehind_days (List[float]):
+            How far behind to look for values
+        embedding_fn (list[Callable]):
```
Comment (HLasse) [inline src/timeseriesflattener/feature_spec_objects.py:1035]:
```suggestion
        embedding_fn (List[Callable]):
```

File: src/timeseriesflattener/feature_spec_objects.py [hunk 2]
```diff
@@ -998,3 +998,70 @@ def create_combinations(self) -> List[OutcomeSpec]:
...
+            A function used for embedding the text. Should take a
+        pandas series of strings and return a pandas dataframe of embeddings.
+        embedding_fn_kwargs (Optional[list[dict]]):
```
Comment (HLasse) [inline src/timeseriesflattener/feature_spec_objects.py:1038]:
```suggestion
        embedding_fn_kwargs (Optional[List[dict]]):
```

File: src/timeseriesflattener/feature_spec_objects.py [hunk 3]
```diff
@@ -998,3 +998,70 @@ def create_combinations(self) -> List[OutcomeSpec]:
...
+            Optional kwargs passed onto the embedding_fn.
+    """
+
+    class Doc:
+        short_description = """Specification for a group of text predictors."""
+
+    embedding_fn: list[Callable] = Field(
```
Comment (HLasse) [inline src/timeseriesflattener/feature_spec_objects.py:1045]:
```suggestion
    embedding_fn: List[Callable] = Field(
```

File: src/timeseriesflattener/feature_spec_objects.py [hunk 4]
```diff
@@ -998,3 +998,70 @@ def create_combinations(self) -> List[OutcomeSpec]:
...
+        description="""A function used for embedding the text. Should take a
+        pandas series of strings and return a pandas dataframe of embeddings.
+        Defaults to: None.""",
+    )
+    embedding_fn_kwargs: Optional[list[dict]] = Field(
```
Comment (HLasse) [inline src/timeseriesflattener/feature_spec_objects.py:1050]:
```suggestion
    embedding_fn_kwargs: Optional[List[dict]] = Field(
```

File: src/timeseriesflattener/feature_spec_objects.py [hunk 5]
```diff
@@ -998,3 +998,70 @@ def create_combinations(self) -> List[OutcomeSpec]:
...
+        default=None,
+        description="""Optional kwargs passed onto the embedding_fn.""",
+    )
+    resolve_multiple_fn: list[Union[Callable, str]] = Field(
```
Comment (HLasse) [inline src/timeseriesflattener/feature_spec_objects.py:1054]:
```suggestion
    resolve_multiple_fn: List[Union[Callable, str]] = Field(
```

**Review by HLasse — APPROVED** [review]
LGTM. There were some type hint issues (using `list` instead of `List` but otherwise feel free to merge

**Pull request merged**